### ISSUE TITLE
Should `Action::Overwrite` in `Assert::subset_eq` overwrite files and dirs in expected root?

### ISSUE THREAD

**Issue by zummenix** [assert-rs/trycmd#230]
Maybe I missing something or using it wrong, please let me know. Current behaviour is that it just reports ok. Seems unexpected when one wants to prepare a snapshot.

**Comment by epage** [comment]
Yes, `Overwrite` should overwrite existing files when used with `subset_eq`.

> Seems unexpected when one wants to prepare a snapshot.

Overwriting means replacing an existing item.  It sounds like you are wanting to write out everything, whether there is an existing item or not.  Is that correct?

**Comment by zummenix** [comment]
> It sounds like you are wanting to write out everything, whether there is an existing item or not. Is that correct?

Yes, that's correct. To clarify, overwriting works for existing items but does not if a snapshot directory is empty. Thinking about it I can workaround this issue by first running a test to create snapshot files.
